src/main/python/pgshovel/streams/utilities.py

```diff
@@ -24,3 +36,43 @@ def __str__(self):
             active=','.join(map(str, active[:self.max])),
             truncated='' if len(active) <= self.max else ',+{0}...'.format(len(active) - self.max),
         )
+
+
+class UnableToPrimeError(Exception):
+    """
+    Raised when an attempt to prime is made, but the priming failed after the
+    max number of messages.
+    """
+    pass
+
+
+def is_start_batch_operation(message):
+    """
+    Is the ``message`` the start operation of a batch of mutations.
+    """
+    value = get_oneof_value(message.batch_operation, 'operation')
+    return type(value) in TRANSACTION_START_EVENT_TYPES
+
+
+def prime_until(check, max_messages, stream):
+    """
+    Attempts to "prime" the ``stream`` by dropping any messages for which
+    ``check`` evaluates to ``False``, up until ``max_messages``.
+    """
+
+    attempts = count(start=1)
```

Comment:
@ivanov your example doesn't run, but I think I see what you're saying.  There is an off by 1 error here where on the `attempt` that exceeds max_messages but `check(message)` returns `True` we raise an exception when we should actually return.

Good catch! I'll fix this.